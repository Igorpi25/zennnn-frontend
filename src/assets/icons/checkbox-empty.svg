<svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" viewBox="0 0 14 14"><title>checkbox-empty</title><path d="M0,0V14H14V0H0ZM12,12H2V2H12V12Z" style="fill:#5a8199"/></svg>