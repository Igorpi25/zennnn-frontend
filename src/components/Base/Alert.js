import { convertToUnit } from '../../util/helpers'

export default {
  name: 'Alert',
  props: {
    value: Boolean,
    // type, can be: primary, error, warn, success, info
    color: {
      type: String,
      default: 'primary',
    },
    icon: {
      type: Boolean,
      default: true,
    },
    close: Boolean,
    text: {
      type: String,
      default: '',
    },
    transition: String,
    minWidth: {
      type: [String, Number],
      default: null,
    },
    maxWidth: {
      type: [String, Number],
      default: 494,
    },
    textColor: String,
    bgColor: String,
    iconColor: String,
  },
  computed: {
    compTextColor () {
      if (this.textColor) return this.textColor
      switch (this.color) {
        case 'error': return 'text-white'
        case 'warn': return 'text-yellow-500'
        case 'success': return 'text-green-500'
        case 'info': return 'text-gray-100'
        default: return 'text-blue-500'
      }
    },
    compBgColor () {
      if (this.bgColor) return this.bgColor
      switch (this.color) {
        case 'error': return 'bg-red-900'
        case 'warn': return 'bg-yellow-500 bg-opacity-10'
        case 'success': return 'bg-green-500 bg-opacity-10'
        case 'info': return 'bg-gray-100 bg-opacity-10'
        default: return 'bg-blue-500 bg-opacity-10'
      }
    },
    compIconColor () {
      if (this.iconColor) return this.iconColor
      switch (this.color) {
        case 'error': return 'text-pink-500'
        case 'info': return 'text-green-500'
        default: return this.compTextColor
      }
    },
  },
  methods: {
    closeAlert () {
      this.$emit('input', false)
    },
    genContent () {
      const children = []
      if (this.icon) {
        children.push(this.$createElement('i', {
          class: ['zi-info text-2xl flex-shrink-0 pr-4', this.compIconColor],
        }))
      }
      const content = this.$slots.default ? this.$slots.default : this.text
      children.push(this.$createElement('div', {
        class: ['flex-grow flex items-center text-sm leading-tight'],
      }, content))
      if (this.close) {
        children.push(this.$createElement('i', {
          class: ['zi-close text-2xl flex-shrink-0 pl-3 cursor-pointer select-none', this.compTextColor],
          on: {
            click: this.closeAlert,
          },
        }))
      }
      return this.$createElement('div', {
        class: [
          'flex rounded-md relative text-white py-3 px-4',
          this.compTextColor,
          this.compBgColor,
        ],
        style: {
          maxWidth: this.maxWidth && !isNaN(Number(this.maxWidth)) ? convertToUnit(this.maxWidth) : this.maxWidth,
          minWidth: this.minWidth && !isNaN(Number(this.minWidth)) ? convertToUnit(this.minWidth) : this.minWidth,
        },
        attrs: {
          role: 'alert',
        },
        directives: [{
          name: 'show',
          value: this.value,
        }],
      }, children)
    },
  },
  render (h) {
    if (this.transition) {
      return h('transition', {
        props: {
          name: this.transition,
        },
      }, [this.genContent()])
    } else {
      return this.value ? [this.genContent()] : null
    }
  },
}
