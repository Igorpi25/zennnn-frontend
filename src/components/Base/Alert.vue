<template>
  <div
    v-if="value"
    :class="['alert', classes]"
  >
    <div class="flex-grow">
      {{ text }}
    </div>
    <div class="flex items-center">
      <a
        v-if="close"
        class="alert_close"
        @click="$emit('input', false)"
      >
        <svg xmlns="http://www.w3.org/2000/svg" width="18" height="18" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/><path d="M0 0h24v24H0z" fill="none"/></svg>
      </a>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Alert',
  props: {
    value: {
      type: Boolean,
      default: false,
    },
    color: {
      type: String,
      default: 'primary',
    },
    close: {
      type: Boolean,
      default: false,
    },
    text: String,
  },
  computed: {
    classes () {
      return `bg-${this.color}`
    },
  },
}
</script>
