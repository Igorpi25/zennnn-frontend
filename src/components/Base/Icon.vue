<script>
import { convertToUnit } from '@/util/helpers'

export default {
  name: 'Icon',
  props: {
    iconName: {
      type: String,
      default: '',
    },
    title: {
      type: [String, Object],
      default: '',
    },
    size: {
      type: [String, Number],
      default: 18,
    },
    color: {
      type: String,
      default: 'currentColor',
    },
  },
  computed: {
    path () {
      return `@/assets/icons/${this.iconName}.svg`
    },
  },
  render (createElement) {
    const h = createElement
    const attrs = {
      xmlns: 'http://www.w3.org/2000/svg',
      width: this.size,
      height: this.size,
      viewBox: '0 0 24 24',
    }
    const size = convertToUnit(this.size)
    if (this.iconName) {
      return h('i', {}, [
        h('img', {
          attrs: {
            src: this.path,
          },
        }),
      ])
    } else if (this.$slots.default[0].text) {
      return h('svg', {
        attrs,
        style: {
          width: size,
          height: size,
          'font-size': size,
          color: this.color,
        },
      },
      [h('title',
        this.title,
      ),
      [h('g', {
        attrs: {
          fill: this.color,
        },
      },
      [h('path', {
        attrs: {
          d: this.$slots.default[0].text,
        },
      })])]])
    }
    return null
    // else {
    //   const comp = this.$slots.default[0]
    //   comp.componentOptions.propsData = {
    //     width: attrs.width,
    //     height: attrs.height,
    //     styles: {
    //       width: size,
    //       height: size,
    //       'font-size': size,
    //     },
    //   }
    //   return comp
    // }
  },
}
</script>
