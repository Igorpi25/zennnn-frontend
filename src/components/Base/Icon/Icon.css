@layer components {
  .icon {
    align-items: center;
    display: inline-flex;
    font-feature-settings: 'liga';
    justify-content: center;
    letter-spacing: normal;
    line-height: 1;
    text-indent: 0;
    vertical-align: middle;
    user-select: none;
  }
  .icon svg {
    text-indent: 0;
    letter-spacing: normal;
    line-height: 1;
    user-select: none;
    fill: currentColor;
    vertical-align: middle;
    width: 1em;
    height: 1em;
  }
}
