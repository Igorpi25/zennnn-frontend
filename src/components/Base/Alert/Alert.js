import {
  h,
  ref,
  computed,
  Transition,
  withDirectives,
  vShow,
  watch,
  cloneVNode,
} from 'vue'

import { ziCloseDelete, ziInfoBig } from '../../../assets/icons'

import { convertToUnit } from '../../../utils/convertToUnit'

import Icon from '../Icon'

import './Alert.css'

export default {
  name: 'Alert',

  props: {
    modelValue: Boolean,
    color: {
      type: String,
      validator: (value) => {
        return [
          'primary',
          'error',
          'warn',
          'success',
          'info',
        ].includes(value)
      },
      default: 'primary',
    },
    infoIcon: {
      type: Boolean,
      default: true,
    },
    close: Boolean,
    text: {
      type: String,
      default: '',
    },
    transition: String,
    minWidth: {
      type: [String, Number],
      default: undefined,
    },
    maxWidth: {
      type: [String, Number],
      default: 494,
    },
    infoIconClass: {
      type: String,
      default: '',
    },
    closeClass: {
      type: String,
      default: '',
    },
    contentClass: {
      type: String,
      default: '',
    },
    containerClass: {
      type: String,
      default: '',
    },
  },

  emits: ['update:modelValue'],

  setup (props, { slots, emit }) {
    const internalValue = ref(props.modelValue)

    watch(() => props.modelValue, (val) => {
      internalValue.value = val
    })

    watch(internalValue, (val) => {
      emit('update:modelValue', val)
    })

    const containerClasses = computed(() => {
      return {
        alert__container: true,
        'alert__container--primary': props.color === 'primary',
        'alert__container--error': props.color === 'error',
        'alert__container--warn': props.color === 'warn',
        'alert__container--success': props.color === 'success',
        'alert__container--info': props.color === 'info',
        'alert__container--closable': props.close,
        [props.containerClass.trim()]: true,
      }
    })

    const styles = computed(() => {
      const styles = {}

      if (props.maxWidth) styles.maxWidth = convertToUnit(props.maxWidth)
      if (props.minWidth) styles.minWidth = convertToUnit(props.minWidth)

      return styles
    })

    const hasIcon = computed(() => {
      return slots.icon || props.infoIcon
    })

    const hasClose = computed(() => {
      return slots.close || props.close
    })

    const close = () => {
      internalValue.value = false
    }

    const genIcon = () => {
      if (slots.icon) {
        return slots.icon()
      } else if (props.infoIcon) {
        return h(Icon, {
          size: 24,
          class: {
            'alert__info-icon': true,
            'alert__info-icon--error': props.color === 'error',
            'alert__info-icon--info': props.color === 'info',
            [props.infoIconClass.trim()]: true,
          },
        }, {
          default: () => ziInfoBig,
        })
      }
      return null
    }

    const genClose = () => {
      const closeChildren = slots.close ? slots.close() : []
      if (closeChildren.length === 1) {
        return cloneVNode(slots.close()[0], { onClick: close })
      } else if (props.close) {
        return h(Icon, {
          size: 24,
          class: {
            alert__close: true,
            [props.closeClass.trim()]: true,
          },
          onClick: close,
        }, {
          default: () => ziCloseDelete,
        })
      }
      return null
    }

    const genContent = () => {
      const children = []

      if (hasIcon.value) {
        children.push(genIcon())
      }

      children.push(h('div', {
        class: {
          alert__content: true,
          [props.contentClass.trim()]: true,
        },
      }, slots.default ? slots.default() : props.text))

      if (hasClose.value) {
        children.push(genClose())
      }

      return withDirectives(
        h('div', {
          class: 'alert',
          style: styles.value,
          role: 'alert',
        }, h('div', {
          class: containerClasses.value,
        }, children)),
        [
          [vShow, internalValue.value],
        ],
      )
    }

    return {
      internalValue,
      genContent,
    }
  },

  render () {
    const content = this.genContent()

    if (this.transition) {
      return h(Transition, {
        name: this.transition,
      }, { default: () => content })
    } else {
      return this.internalValue ? content : undefined
    }
  },
}
