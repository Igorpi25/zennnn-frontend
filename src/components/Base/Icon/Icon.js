import { h, computed } from 'vue'

import { convertToUnit } from '../../../utils/convertToUnit'

import './Icon.css'

const SIZE_CLASS_MAP = {
  small: '18px',
  base: '24px',
  large: '32px',
}

export default {
  name: 'Icon',

  props: {
    tag: {
      type: String,
      default: 'i',
    },
    size: {
      type: [Number, String],
      default: undefined,
    },
    small: Boolean,
    large: Boolean,
    base: {
      type: Boolean,
      default: true,
    },
  },

  setup (props, { attrs, slots }) {
    const tag = computed(() => props.tag || 'i')

    const fontSize = computed(() => {
      return props.size
        ? convertToUnit(props.size)
        : props.small
          ? SIZE_CLASS_MAP.small
          : props.large
            ? SIZE_CLASS_MAP.large
            : props.base
              ? SIZE_CLASS_MAP.base
              : undefined
    })

    const genSvg = () => {
      const node = slots.default ? slots.default() : []
      const children = node[0] && node[0].children
      const icon = typeof children === 'string' ? children : undefined

      if (!icon) return undefined

      const svgData = {
        xmlns: 'http://www.w3.org/2000/svg',
        viewBox: '0 0 24 24',
        height: fontSize.value || '24',
        width: fontSize.value || '24',
        role: 'img',
        'aria-hidden': true,
      }
      if (fontSize.value) {
        svgData.style = {
          fontSize: fontSize.value,
          height: fontSize.value,
          width: fontSize.value,
        }
      }

      return h('svg', svgData, h('g', {
        fill: 'currentColor',
      }, h('path', {
        d: icon,
      })))
    }

    const classes = { icon: true }

    const clickable = attrs.onClick

    return () => {
      return h(tag.value, {
        class: classes,
        role: clickable ? 'button' : undefined,
      }, genSvg())
    }
  },
}
