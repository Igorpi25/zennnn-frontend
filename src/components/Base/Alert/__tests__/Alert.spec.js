import { h } from 'vue'
import { mount } from '@vue/test-utils'
import Alert from '../Alert'
import Icon from '../../Icon'
import { ziCheckFill } from '../../../../assets/icons'

describe('Alert.js', () => {
  let mountFunction

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return mount(Alert, {
        ...options,
      })
    }
  })

  // Render

  it('should render component with default slot and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
      },
      slots: {
        default: () => 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with text prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with "error" color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        color: 'error',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with "warn" color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        color: 'warn',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with "success" color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        color: 'success',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with "info" color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        color: 'info',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with "primary" color and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        color: 'primary',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with close prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        close: true,
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom class props and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        close: true,
        closeClass: 'custom-close-class',
        infoIconClass: 'custom-info-icon-close',
        contentClass: 'content-custom-class',
        class: 'custom-class',
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with icon slot and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        close: true,
        text: 'Alert',
      },
      slots: {
        icon: () => h(Icon, {
          base: true,
          class: 'text-blue-500 pr-4',
        }, { default: () => ziCheckFill }),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component without icon and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        infoIcon: false,
        text: 'Alert',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with close slot and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        text: 'Alert',
      },
      slots: {
        close: () => h('button', { class: 'custom-close' }, 'Close'),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit false modelValue on close icon click', async () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        close: true,
        text: 'Alert',
      },
    })

    await wrapper.find('.alert__close').trigger('click')

    expect(wrapper.emitted()['update:modelValue'][0]).toEqual([false])
  })

  it('should toggle model value', async () => {
    const wrapper = mountFunction({
      slots: {
        default: 'Alert',
      },
    })

    expect(wrapper.html()).toBe('<!---->')

    await wrapper.setProps({ modelValue: true })

    expect(wrapper.classes('alert')).toBe(true)
  })

  it('should unset default maxWidth prop', async () => {
    const wrapper = mountFunction({
      props: {
        modelValue: true,
        maxWidth: 'none',
      },
    })

    expect(wrapper.element.style.maxWidth).toBe('none')
  })
})
