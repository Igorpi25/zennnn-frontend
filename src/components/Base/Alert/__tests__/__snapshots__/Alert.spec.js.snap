// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert.js should render component with "error" color and match snapshot 1`] = `
<div class="alert" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--error"><i class="icon alert__info-icon alert__info-icon--error"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.575 6.088a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.1c0-.658.56-1.191 1.25-1.191s1.25.533 1.25 1.191v5.87c0 .659-.56 1.192-1.25 1.192s-1.25-.533-1.25-1.191v-5.87zM1.775 12.042c0-5.964 4.836-10.8 10.8-10.8 5.965 0 10.8 4.836 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zm10.8-9.2a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with "info" color and match snapshot 1`] = `
<div class="alert" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--info"><i class="icon alert__info-icon alert__info-icon--info"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.575 6.088a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.1c0-.658.56-1.191 1.25-1.191s1.25.533 1.25 1.191v5.87c0 .659-.56 1.192-1.25 1.192s-1.25-.533-1.25-1.191v-5.87zM1.775 12.042c0-5.964 4.836-10.8 10.8-10.8 5.965 0 10.8 4.836 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zm10.8-9.2a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with "primary" color and match snapshot 1`] = `
<div class="alert" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--primary"><i class="icon alert__info-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.575 6.088a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.1c0-.658.56-1.191 1.25-1.191s1.25.533 1.25 1.191v5.87c0 .659-.56 1.192-1.25 1.192s-1.25-.533-1.25-1.191v-5.87zM1.775 12.042c0-5.964 4.836-10.8 10.8-10.8 5.965 0 10.8 4.836 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zm10.8-9.2a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with "success" color and match snapshot 1`] = `
<div class="alert" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--success"><i class="icon alert__info-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.575 6.088a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.1c0-.658.56-1.191 1.25-1.191s1.25.533 1.25 1.191v5.87c0 .659-.56 1.192-1.25 1.192s-1.25-.533-1.25-1.191v-5.87zM1.775 12.042c0-5.964 4.836-10.8 10.8-10.8 5.965 0 10.8 4.836 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zm10.8-9.2a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with "warn" color and match snapshot 1`] = `
<div class="alert" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--warn"><i class="icon alert__info-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.575 6.088a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.1c0-.658.56-1.191 1.25-1.191s1.25.533 1.25 1.191v5.87c0 .659-.56 1.192-1.25 1.192s-1.25-.533-1.25-1.191v-5.87zM1.775 12.042c0-5.964 4.836-10.8 10.8-10.8 5.965 0 10.8 4.836 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zm10.8-9.2a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with close prop and match snapshot 1`] = `
<div class="alert" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--primary alert__container--closable"><i class="icon alert__info-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.575 6.088a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.1c0-.658.56-1.191 1.25-1.191s1.25.533 1.25 1.191v5.87c0 .659-.56 1.192-1.25 1.192s-1.25-.533-1.25-1.191v-5.87zM1.775 12.042c0-5.964 4.836-10.8 10.8-10.8 5.965 0 10.8 4.836 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zm10.8-9.2a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div><i class="icon alert__close"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.677 10.83L9.14 7.294 7.727 8.71l3.536 3.535-3.536 3.536 1.414 1.414 3.536-3.536 3.535 3.536 1.415-1.414-3.536-3.536 3.536-3.535-1.415-1.415-3.535 3.536z"></path>
        </g>
      </svg></i>
  </div>
</div>
`;

exports[`Alert.js should render component with close slot and match snapshot 1`] = `
<div class="alert" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--primary"><i class="icon alert__info-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.575 6.088a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.1c0-.658.56-1.191 1.25-1.191s1.25.533 1.25 1.191v5.87c0 .659-.56 1.192-1.25 1.192s-1.25-.533-1.25-1.191v-5.87zM1.775 12.042c0-5.964 4.836-10.8 10.8-10.8 5.965 0 10.8 4.836 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zm10.8-9.2a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div><button class="custom-close">Close</button>
  </div>
</div>
`;

exports[`Alert.js should render component with custom class props and match snapshot 1`] = `
<div class="alert custom-class" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--primary alert__container--closable"><i class="icon alert__info-icon custom-info-icon-close"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.575 6.088a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.1c0-.658.56-1.191 1.25-1.191s1.25.533 1.25 1.191v5.87c0 .659-.56 1.192-1.25 1.192s-1.25-.533-1.25-1.191v-5.87zM1.775 12.042c0-5.964 4.836-10.8 10.8-10.8 5.965 0 10.8 4.836 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zm10.8-9.2a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content content-custom-class">Alert</div><i class="icon alert__close custom-close-class"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.677 10.83L9.14 7.294 7.727 8.71l3.536 3.535-3.536 3.536 1.414 1.414 3.536-3.536 3.535 3.536 1.415-1.414-3.536-3.536 3.536-3.535-1.415-1.415-3.535 3.536z"></path>
        </g>
      </svg></i>
  </div>
</div>
`;

exports[`Alert.js should render component with default slot and match snapshot 1`] = `
<div class="alert" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--primary"><i class="icon alert__info-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.575 6.088a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.1c0-.658.56-1.191 1.25-1.191s1.25.533 1.25 1.191v5.87c0 .659-.56 1.192-1.25 1.192s-1.25-.533-1.25-1.191v-5.87zM1.775 12.042c0-5.964 4.836-10.8 10.8-10.8 5.965 0 10.8 4.836 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zm10.8-9.2a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component with icon slot and match snapshot 1`] = `
<div class="alert" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--primary alert__container--closable"><i class="icon text-2xl text-blue-500 pr-4"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24" width="24" role="img" aria-hidden="true">
        <g fill="currentColor">
          <path d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm5.045 5.719a1.25 1.25 0 01.941 2.11l-5.34 5.636a1.25 1.25 0 01-1.804.01L7.273 11.79a1.25 1.25 0 011.797-1.738l2.659 2.744 4.443-4.688c.237-.25.553-.382.873-.39z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div><i class="icon alert__close"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.677 10.83L9.14 7.294 7.727 8.71l3.536 3.535-3.536 3.536 1.414 1.414 3.536-3.536 3.535 3.536 1.415-1.414-3.536-3.536 3.536-3.535-1.415-1.415-3.535 3.536z"></path>
        </g>
      </svg></i>
  </div>
</div>
`;

exports[`Alert.js should render component with text prop and match snapshot 1`] = `
<div class="alert" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--primary"><i class="icon alert__info-icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" height="24px" width="24px" role="img" aria-hidden="true" style="font-size: 24px; height: 24px; width: 24px;">
        <g fill="currentColor">
          <path d="M12.575 6.088a1.251 1.251 0 000 2.5 1.251 1.251 0 000-2.5zm-1.25 5.1c0-.658.56-1.191 1.25-1.191s1.25.533 1.25 1.191v5.87c0 .659-.56 1.192-1.25 1.192s-1.25-.533-1.25-1.191v-5.87zM1.775 12.042c0-5.964 4.836-10.8 10.8-10.8 5.965 0 10.8 4.836 10.8 10.8 0 5.965-4.835 10.8-10.8 10.8-5.964 0-10.8-4.835-10.8-10.8zm10.8-9.2a9.2 9.2 0 100 18.4 9.2 9.2 0 000-18.4z"></path>
        </g>
      </svg></i>
    <div class="alert__content">Alert</div>
  </div>
</div>
`;

exports[`Alert.js should render component without icon and match snapshot 1`] = `
<div class="alert" style="max-width: 494px;" role="alert">
  <div class="alert__container alert__container--primary">
    <div class="alert__content">Alert</div>
  </div>
</div>
`;
